Trippy/View Controllers/Itinerary Tab/EditingItineraryViewController.m

```diff
@@ -62,10 +80,34 @@ - (void)viewDidLoad {
     self.editView.layer.shadowOpacity = VIEW_SHADOW_OPACITY;
     self.editView.layer.shadowRadius = VIEW_SHADOW_RADIUS;
     
-    self.departureDatePicker.date = self.baseItinerary.departureTime;
+    self.editView.backgroundColor = [UIColor whiteColor];
+    
+    [self.loadingIndicator setHidesWhenStopped:YES];
+    self.cacheHandler = [[CacheDataHandler alloc] init];
+    self.cacheHandler.delegate = self;
 }
 
 - (IBAction)tapReroute:(id)sender {
+    [self.loadingIndicator startAnimating];
+    NSString *matrixUrl = [MapUtils generateMatrixApiUrl:self.mutableItinerary.sourceCollection
+                                            origin:self.mutableItinerary.originLocation
+                                     departureTime:self.mutableItinerary.departureTime];
+    [[MapsAPIManager shared] getRouteMatrixWithCompletion:matrixUrl completion:^(NSDictionary * _Nonnull response, NSError * _Nonnull) {
```

Comment:
Is this right? Both the response and error result are nonnull? I think, given the design you've been rolling with, both should be nullable, and you check the one or the other. I can see returning an empty nonnull response dictionary possibly, but the error should definitely be nullable (and you should add error checking to this function too).